### PR TITLE
More debugging around (1) retries and (2) query-monitoring

### PR DESCRIPTION
**Pull request by agam**
- Added `QueryMonitoringThreshold` as a configurable parameter passed in
- Log up to 5k of the response body

### CODE REVIEW

**Review by mtoader — APPROVED** [review]

File: retry.go [hunk 1]
```diff
@@ -247,9 +258,10 @@ func (r *retryHTTP) execute() (res *http.Response, err error) {
 				// This is currently used for Snowflake login. The caller must generate an error object based on HTTP status.
 				break
 			}
+			sampledResponseBody = getBodyTrace(res)
```
Comment (mhseiden) [inline retry.go:261]: Is there a specific reason we only want to pull partial error response bodies?

Comment (mtoader) [reply retry.go:261]: They are biggish, mostly html and not that rare.

Comment (mhseiden) [reply retry.go:261]: Ok. 5k still seems large if this isn't rare... but it's easy enough to turn off.

Comment (agam) [reply retry.go:261]: Because 
1. I'm dumping unconditionally
2. I'm nervous about dumping _huge_ response bodies in the logs

**Pull request merged**